19_shamshaw/_targets.R

```diff
@@ -15,81 +16,74 @@ list(
   # Data > Data from National Project > Streamflow > Drought Summaries
   tar_target(events_crb_jd_1980_2020,
              read_csv("data/weibull_jd_30d_wndw_Drought_Properties.csv") %>%
-               transform(StaID = as.character(StaID))%>%
-               mutate(across(c(start, end, previous_end), ~as.Date(.x, '%m/%d/%y')))), 
-  
+               transform(StaID = as.character(StaID)) %>%
+               mutate(across(c(start, end, previous_end), ~as.Date(.x, '%m/%d/%y')))
+             ), 
+
   # Preliminary version of file which includes 2021 data
   tar_target(events_crb_jd_1980_2021,
              read_csv("data/weibull_jd_30d_wndw_Drought_Properties_2021.csv") %>%
                transform(StaID = as.character(StaID))%>%
-               mutate(across(c(start, end, previous_end), ~as.Date(.x, '%m/%d/%y')))), 
+               mutate(across(c(start, end, previous_end), ~as.Date(.x, '%m/%d/%y')))
+             ), 
+  # Combine data long format
+  tar_target(
+    crb_events,
+    bind_rows(events_crb_jd_1980_2020, events_crb_jd_1980_2021) %>%
+      distinct() %>%
+      transform(StaID = as.character(StaID))%>%
+      mutate(across(c(start, end, previous_end), ~as.Date(.x, '%m/%d/%y')))
+  ),
```

Comment:
Ah, I think we'd be getting duplicate events in here as relying on `distinct()` here wouldn't remove duplicates since the 2021 file will have slightly different versions of 1980-2020 events. So, to combine the `events_crb_jd_1980_2021` should be filtered first to events starting form `2020-04-01` or later. I think modifying to:
`bind_rows(events_crb_jd_1980_2020, events_crb_jd_1980_2021[which(events_crb_jd_1980_2021$start >= "2020-04-01"),]) %>%`
would do the trick

Reply:
that made a big difference!